ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+/**
+ * @module DashboardQuickActionsCard
+ * DashboardQuickActionsCard component allows users to see a list of secrets engines filtered by
+ * kv, pki and database and perform certain actions based on the type of secret engine selected
+ *
+ * @example
+ * ```js
+ * <DashboardQuickActionsCard @secretsEngines={{@model.secretsEngines}} />
+ * ```
+ * @param {array}
+ */
+
+const ENGINE_TYPE = ['pki', 'kv', 'database'];
+
+export default class DashboardQuickActionsCard extends Component {
+  @service router;
+
+  @tracked selectedEngine;
+  @tracked selectedAction;
+  @tracked selectedEngineName;
```

Comment:
This is used to pass to `transitionTo` right? Naming it `mountPath` might be a little clearer since engine name can be confused with the engine type